test/livebook_web/live/session_live_test.exs

```diff
@@ -212,6 +216,45 @@ defmodule LivebookWeb.SessionLiveTest do
 
       assert %{input_values: %{"input1" => "line\nline"}} = Session.get_data(session.pid)
     end
+
+    test "form input changes are reflected only in local LV data",
+         %{conn: conn, session: session, test: test} do
+      section_id = insert_section(session.pid)
+
+      Process.register(self(), test)
+
+      form_control = %{
+        type: :form,
+        ref: :form_ref,
+        destination: test,
+        fields: [
+          name: %{
+            ref: :reference,
+            id: "input1",
+            type: :text,
+            label: "Name",
+            default: "initial",
+            destination: test
+          }
+        ],
+        submit: "Send",
+        report_changes: %{},
+        reset_on_submit: []
+      }
+
+      insert_cell_with_output(session.pid, section_id, {:control, form_control})
+
+      {:ok, view, _} = live(conn, "/sessions/#{session.id}")
+
+      view
+      |> element(~s/[data-element="outputs-container"] form/)
+      |> render_change(%{"value" => "sherlock"})
+
+      # The new value is on the page
+      assert render(view) =~ "sherlock"
+      # but it's not reflected in the synchronized session data
+      assert %{input_values: %{"input1" => "initial"}} = Session.get_data(session.pid)
+    end
```

Comment:
I wonder if there is a way to test when we send the data over the wire back to Kino?

Reply:
The messages are sent to input/control `:destination`, which we already set to the testing process. I've just added assertions on the messages, let me know if that's what you had in mind :)